postgres/postgres.go

```diff
@@ -3,18 +3,56 @@ package postgres
 import (
 	"database/sql"
 	"fmt"
-	_ "github.com/lib/pq"
-	sqlds "github.com/whyrusleeping/sql-datastore"
+
+	_ "github.com/lib/pq" //postgres driver
+	"github.com/whyrusleeping/sql-datastore"
 )
 
-type PostgresOptions struct {
+// Options are the postgres datastore options, reexported here for convenience.
+type Options struct {
 	Host     string
 	User     string
 	Password string
 	Database string
 }
 
-func (opts *PostgresOptions) Create() (*sqlds.SQLDatastore, error) {
+type Queries struct {
+}
+
+func (Queries) Delete() string {
+	return `DELETE FROM blocks WHERE key = $1`
+}
+
+func (Queries) Exists() string {
+	return `SELECT exists(SELECT 1 FROM blocks WHERE key=$1)`
+}
+
+func (Queries) Get() string {
+	return `SELECT data FROM blocks WHERE key = $1`
+}
+
+func (Queries) Put() string {
+	return `INSERT INTO blocks (key, data) SELECT $1, $2 WHERE NOT EXISTS ( SELECT key FROM blocks WHERE key = $1)`
+}
+
+func (Queries) Query() string {
+	return `SELECT key, data FROM blocks`
```

Comment:
The order by clause throws errors in the tests: 
ds_test.go:136: pq: syntax error at or near "WHERE"
ds_test.go:379: pq: syntax error at or near "WHERE"

and errors when using the ipfs daemon:
13:51:16.640 ERROR  providers: error reading providers:  pq: syntax error at or near "WHERE" providers.go:263
13:51:17.685 ERROR  providers: error adding new providers:  pq: syntax error at or near "WHERE" providers.go:258

Reply:
That's probably because it's applying a prefix and `ORDER BY` must come after `WHERE` (as far as I know).

I don't think it's an issue as we don't currently make any guarantees around ordering (unless an order is specified but you apply that separately). Ideally the database would handle basic ordering requests but we can deal with that later.

Reply:
That appears to be the issue, ordering of the results is working in the [tests](https://github.com/whyrusleeping/sql-datastore/blob/7cda9c2c07e1f887ebfce66917d8ace81ef86834/ds_test.go#L156) when the query is qualified with an [order](https://github.com/ipfs/go-datastore/blob/master/query/order.go).